src/Core/src/Credentials/IniFileProvider.php

```diff
@@ -92,6 +93,16 @@ private function getCredentialsFromProfile(array $profilesData, string $profile,
             return $this->getCredentialsFromRole($profilesData, $profileData, $profile, $circularCollector);
         }
 
+        if (isset($profileData[IniFileLoader::KEY_SSO_START_URL])) {
+            if (class_exists(SsoClient::class)) {
+                return $this->getCredentialsFromLegacySso($profileData, $profile);
+            }
+
+            $this->logger->warning('The profile "{profile}" contains SSO (legacy) config but the "async-aws/sso" package is not installed.', ['profile' => $profile]);
```

Comment:
For a better DX, this deserve a `Run "composer require async-aws/sso" to add the missing dependency` (or something like that

Reply:
I adjusted the log message